src/package1/CustomArrayList.java

```diff
@@ -100,10 +100,13 @@ public boolean addAll(Collection<? extends String> c) {
             String[] a = Arrays.copyOf(elements, (int) (elements.length + c.size() * 1.5));
             elements = a;
         }
+        int numElementC = 0;
 
-        System.arraycopy(elements, size, c, 0, c.size() - 1);
-        size=+c.size();
-
+        for (String s: c){
+            elements[size + numElementC] = s;
+            numElementC++;
+        }
+        size = size + c.size();
```

Comment:
It seems to be not optimized. Let's imagine you have 1 000 000 elements. To add all these elements you need first to iterate all incoming collection and then add elemebs one by one.

Much better would be to get an array from the collection
**String[] updatedFoos = coll.toArray(String[]::new);**

and then copy all elements with System.arraycopy()